src/contexts/APIRouterContext/Provider.tsx

```diff
@@ -0,0 +1,157 @@
+
+  // This ref gets called after the user submits their secretPhrase
+  const afterSecretCB = useRef<(secretPhrase: string) => Promise<void> | undefined>();
+
+  const _handleSendJUP = useCallback(
+    async (tx: IUnsignedTransaction, secretPhrase: string) => {
+      tx.secretPhrase = secretPhrase;
+
+      const result = await sendJUP(tx);
+
+      console.log("send result:", result);
+
+      if (!result) {
+        enqueueSnackbar(messageText.transaction.failure, { variant: "error" });
+        return;
+      }
+
+      enqueueSnackbar(messageText.transaction.success, { variant: "success" });
+    },
+    [enqueueSnackbar]
+  );
+
+  const handleSendJUP = useCallback(
+    async (toAddress: string, amount: string): Promise<true | undefined> => {
+      // TODO: validate amount at the input layer, or here or somewhere smort
+
+      if (accountRs === undefined || handleFetchAccountIDFromRS === undefined) {
+        // TODO: error reporting this properly
+        return;
+      }
+
+      // make sure address is valid
+      if (!isValidAddress(toAddress)) {
+        return;
+      }
+
+      const recipientAccountId = await handleFetchAccountIDFromRS(toAddress);
+      const tx: IUnsignedTransaction = {
+        senderPublicKey: publicKey, // publicKey from useAccount() hook
+        senderRS: accountRs, // accountRs from useAccount() hook
+        // sender: "123", // required in some situations?
+        feeNQT: standardFee,
+        version: 1,
+        phased: false,
+        type: 0,
+        subtype: 0,
+        attachment: { "version.OrdinaryPayment": 0 },
+        amountNQT: amount, // TODO: use converter function, but for now it's nice for cheaper testing
+        recipientRS: toAddress,
+        recipient: recipientAccountId,
+        ecBlockHeight: 0, // must be included
+        deadline: standardDeadline,
+        timestamp: Math.round(Date.now() / 1000) - JUPGenesisTimestamp, // Seconds since Genesis. sets the origination time of the tx (since broadcast can happen later).
+        secretPhrase: "",
+      };
+
+      afterSecretCB.current = _handleSendJUP.bind(null, tx);
+      setRequestUserSecret(true);
+
+      return true;
+    },
+    [_handleSendJUP, accountRs, handleFetchAccountIDFromRS, publicKey]
+  );
+
+  const handleSecretEntry = useCallback((secretInput) => {
+    setUserSecretInput(secretInput);
+  }, []);
+
+  const handleCloseSeedCollection = useCallback(
+    (isSuccess: boolean) => {
+      let messageVariant: VariantType = "warning";
+      setRequestUserSecret(false);
+
+      if (isSuccess) {
+        messageVariant = "success";
+      }
+
+      enqueueSnackbar(messageText.transaction.cancel, { variant: messageVariant });
+    },
+    [enqueueSnackbar]
+  );
+
+  const handleSubmitSecret = useCallback(async () => {
+    let result;
+    try {
+      if (afterSecretCB.current !== undefined) {
+        result = await afterSecretCB.current(userSecretInput);
+      }
```

Comment:
if this logic doesn't get triggered it could be hard to locate the problem, consider a console.log/error after this logic?

Reply:
yeah, several approaches could be taken depending on style.

if(... === undefined) throw new Error(...)

or, as you said, adding a console statement.

i don't really care which way it goes